internal/reporter.go

```diff
@@ -18,11 +19,20 @@ type reporter struct {
 }
 
 // NewReporter create a metrics Reporter
-func NewReporter(server string, token string) Reporter {
+func NewReporter(server string, token string, timeout time.Duration, tlsConfigOptions *tls.Config) Reporter {
```

Comment:
Have you considered making NewReporter() look like this:
```func NewReporter(server string, token string, client *http.Client) Reporter```
Since configuration has the tls.config and the timeout you could add a ```NewClient() *http.Client``` method to configuration.

Reply:
I hadn't! Thanks, I really like this suggestion, since it reduces the number of arguments to NewReporter(). Your way of describing it made it easy to implement. 😊